### PR TITLE
Simplify `scattering1d`

### LINKED ISSUE #852: Unnecessary arguments in `core/scattering1d/scattering1d`
Not a big priority, but an easy simplification. The current prototype of `scattering` is: (phew!)

```python
scattering1d(x, pad, unpad, backend, log2_T, psi1, psi2, phi, pad_left=0,
        pad_right=0, ind_start=None, ind_end=None, oversampling=0,
        max_order=2, average=True, vectorize=False, out_type='array')
```

This didn't come to my attention while reviewing #673 , but the 5th input parameter, `log2_T`, is unnecessary: it is already available as `phi["j"]`. So we could just remove it, and put `log2_T = phi["j"]` at the top of the function.

Thoughts?

### PR DESCRIPTION
**Pull request by lostanlen**
Fixes #852 and does more :)

This PR removes five input arguments from the core `scattering1d` function while respecting backwards compatibility. 
It will add a few lines of code inside `scattering1d` but will make calls for the 1D core way more lightweight and readable.

Here's how i did it:

```python
pad = backend.pad
unpad = backend.unpad

# compute the padding quantities:
N = x.shape[-1]
N_pad = phi[0].shape[-1]
pad_left, pad_right = compute_padding(N, N_pad)

# compute start and end indices
log2_T = phi["j"]
J = max(max(map(itemgetter("j"), psi1)), max(map(itemgetter("j"), psi2)))
ind_start, ind_end = compute_border_indices(log2_T, J, pad_left, pad_left+N)
```

I only had to modify `compute_padding` so that it takes `N, N_pad` as argument instead of `J_pad,N`. But it's equivalent since`N_pad=2**J_pad`.

This first commit is a fully working example, ready for review. After it's accepted, i'll take care of cleaning up extra keyword arguments from core and call sites (TF/PT/NP frontends) and will confirm the new prototype of `scattering1d` and `compute_padding`

### CODE REVIEW

File: kymatio/scattering1d/frontend/base_frontend.py [hunk 1]
```diff
@@ -71,11 +71,6 @@ def build(self):
         J_max_support = int(np.floor(np.log2(3 * self.N - 2)))
         self.J_pad = min(int(np.ceil(np.log2(self.N + 2 * min_to_pad))),
                          J_max_support)
-        # compute the padding quantities:
-        self.pad_left, self.pad_right = compute_padding(self.J_pad, self.N)
-        # compute start and end indices
-        self.ind_start, self.ind_end = compute_border_indices(
-            self.log2_T, self.J, self.pad_left, self.pad_left + self.N)
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/base_frontend.py:78]: I prefer this version of doing the padding calculation. I don't think this calculation should be put into core.

**Review by MuawizChaudhary — CHANGES_REQUESTED** [review]
Not sure I support this PR. @janden @eickenberg thoughts?

File: kymatio/scattering1d/frontend/base_frontend.py (hunk 1 above)
Comment (lostanlen) [reply kymatio/scattering1d/frontend/base_frontend.py:78]: why?

**Comment by lostanlen** [comment]
i'm struggling with name resolutions and module imports but this is clearly doable ([a9d2db6](https://github.com/kymatio/kymatio/pull/855/commits/a9d2db67b2590c223e96b101e379b967d8336630))

The question is should we do it.

**Pull request closed without merging**